zrml/prediction-markets/src/lib.rs

```diff
@@ -2221,6 +2167,12 @@ mod pallet {
         /// Clears this market from being stored for automatic resolution.
         fn clear_auto_resolve(market_id: &MarketIdOf<T>) -> Result<(u32, u32), DispatchError> {
             let market = <zrml_market_commons::Pallet<T>>::market(market_id)?;
+            // If there's no dispute mechanism, this function is noop. FIXME This is an
+            // anti-pattern, but it makes benchmarking easier.
+            let dispute_mechanism = match market.dispute_mechanism {
+                Some(ref result) => result,
+                None => return Ok((0, 0)),
+            };
```

Comment:
This won't work because the `return` statement only applies to the closure, not the entire function. I don't think Rust has a one-line solution for this.